pkg/apis/core/v1beta1/types_shoot.go

```diff
@@ -567,6 +568,10 @@ type DNSProvider struct {
 	// provider. When not specified, the Gardener will use the cloud provider credentials referenced
 	// by the Shoot and try to find respective credentials there (primary provider only). Specifying this field may override
 	// this behavior, i.e. forcing the Gardener to only look into the given secret.
+	//
+	// Deprecated: This field is deprecated and will be forbidden starting from Kubernetes 1.35. Please use `CredentialsRef` instead.
+	// Until removed, this field is synced with the `CredentialsRef` field when it refers to a secret.
+	// TODO(vpnachev): Remove this field once support for Kubernetes 1.34 is dropped.
 	// +optional
 	SecretName *string `json:"secretName,omitempty" protobuf:"bytes,3,opt,name=secretName"`
```

Comment:
I think we can do this so the TODO does not end up in the API reference.
```suggestion
// +optional
SecretName *string `json:"secretName,omitempty" protobuf:"bytes,3,opt,name=secretName"` // TODO(vpnachev): Remove this field once support for Kubernetes 1.34 is dropped.
```